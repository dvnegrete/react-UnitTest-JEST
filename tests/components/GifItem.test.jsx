import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', ()=>{
  const title = 'Pruebas de Gif';
  const url = 'https://gift-pruebas.com.mx/prueba.jpeg';
  
  test('should has match with the snapshot', ()=>{
    const { container } =  render(<GifItem title={ title } url={ url }/>);
    expect(container).toMatchSnapshot();
  });

  test('should show img with URL and ALT right', ()=>{
    render(<GifItem title={ title } url={ url }/>);
    //screen.debug();
    // expect( screen.getByRole('img').src).toBe(url);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  
  test('shoul title in element p', ()=>{
    render(<GifItem title={ title } url={ url }/>);
    expect( screen.getByText(title) ).toBeTruthy();

  });

});