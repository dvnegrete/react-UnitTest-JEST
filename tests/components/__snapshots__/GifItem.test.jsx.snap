// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should has match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pruebas de Gif"
      src="https://gift-pruebas.com.mx/prueba.jpeg"
    />
    <p>
      Pruebas de Gif
    </p>
  </div>
</div>
`;
